src_c/math.c

```diff
@@ -1287,6 +1293,77 @@ vector_scale_to_length(pgVector *self, PyObject *length)
     Py_RETURN_NONE;
 }
 
+static int
+_vector_move_towards_helper(Py_ssize_t dim, double *origin_coords, double *target_coords, double distance)
+{
+    Py_ssize_t i;
+    double delta[VECTOR_MAX_SIZE];
+    double dist;
+
+    for (i = 0; i < dim; ++i)
+        delta[i] = origin_coords[i] - target_coords[i];
+
+    /* Get magnitude of Vector */
+    dist = sqrt(_scalar_product(delta, delta, dim));
+
+    if (dist == 0)
+        return 1;
+
+    if (dist <= distance)
+    {
+        /* Return target Vector */
+        for (i = 0; i < dim; ++i)
+            origin_coords[i] = target_coords[i];
+        return 1;
+    }
+
+    for (i = 0; i < dim; ++i)
+        origin_coords[i] = origin_coords[i] - delta[i] / dist * distance;
+
+    return 1;
+}
+
+static PyObject *
+vector_move_towards(pgVector *self, PyObject *args)
+{
+    double *target_coords[VECTOR_MAX_SIZE];
+    double distance;
+    pgVector *ret;
+
+    if (!PyArg_ParseTuple(args, "Od:move_towards", &target_coords, &distance))
+        return NULL;
+
+    ret = (pgVector *)pgVector_NEW(self->dim);
+    if (ret == NULL)
+        return NULL;
+
+    for (i = 0; i < self->dim; ++i)
+        ret->coords[i] = self->coords[i];
+
+    if (!_vector_move_towards_helper(self->dim, ret->coords, target_coords, distance))
+    {
+        Py_DECREF(ret);
+        return NULL;
+    }
+    return (PyObject *)ret;
+}
+
+static PyObject *
+vector_move_towards_ip(pgVector *self, PyObject *args)
+{
+    Py_ssize_t i;
+    double *target_coords[VECTOR_MAX_SIZE];
+    double distance;
+    
+    if (!PyArg_ParseTuple(args, "Od:move_towards_ip", &target_coords, &distance))
```

Comment:
*We still need proper arg handling here* 😅 
See my comment above for code and explanation